pkg/cmd/server/api/validation/validation.go

```diff
@@ -21,6 +21,30 @@ import (
 	cmdflags "github.com/openshift/origin/pkg/cmd/util/flags"
 )
 
+func ValidateExternalizableString(s api.ExternalizableString, fieldPath *field.Path) ValidationResults {
+	validationResults := ValidationResults{}
+	methods := 0
+	if len(s.Value) > 0 {
+		methods++
+	}
+	if len(s.File) > 0 {
+		methods++
+		validationResults.AddErrors(ValidateFile(s.File, fieldPath.Child("file"))...)
+	}
+	if len(s.Env) > 0 {
+		methods++
+	}
+	if methods > 1 {
+		validationResults.AddErrors(field.Invalid(fieldPath, "", "only one of value, file, and env can be specified"))
+	}
+
+	if len(s.KeyFile) > 0 {
+		validationResults.AddErrors(ValidateFile(s.KeyFile, fieldPath.Child("keyFile"))...)
```

Comment:
We don't have a validator that checks that its actually a key?  That feels like a hole.  How about an issue, maybe one for @sgallagher?